sbt-coursier/src/main/scala-2.10/coursier/ToSbt.scala

```diff
@@ -1,90 +1,103 @@
 package coursier
 
 import java.util.GregorianCalendar
+import java.util.concurrent.ConcurrentHashMap
 
 import coursier.maven.MavenSource
-
 import sbt._
 
 object ToSbt {
 
-  def moduleId(dependency: Dependency): sbt.ModuleID =
+  private def caching[K, V](f: K => V): K => V = {
+
+    val cache = new ConcurrentHashMap[K, V]
```

Comment:
Just a suggestion, but I think you could use the [TrieMap](http://www.scala-lang.org/api/current/scala/collection/concurrent/TrieMap.html) from the Scala std-lib here. The `getOrElseUpdate` should have the same behaviour as the code below.